src/components/__tests__/DynamicWidgets.js

```diff
@@ -0,0 +1,178 @@
+import { createLocalVue, mount } from '@vue/test-utils';
+import DynamicWidgets from '../DynamicWidgets';
+import { __setState } from '../../mixins/widget';
+import { plugin } from '../../plugin';
+jest.mock('../../mixins/widget');
+
+it('renders all children without state', () => {
+  const localVue = createLocalVue();
+
+  localVue.use(plugin);
+
+  __setState(null);
+
+  const wrapper = mount(DynamicWidgets, {
+    localVue,
+    propsData: {
+      transformItems: items => items,
+    },
+    slots: {
+      default: `
+      <ais-refinement-list attribute="test1"/>
+      <ais-refinement-list attribute="test2"/>
+      <ais-panel>
+        <ais-hierarchical-menu :attributes="['test3', 'test3']" />
+      </ais-panel>
+      `,
+    },
+  });
+
+  // the inner widgets don't render anything because state is falsy, but they are mounted
+  expect(wrapper.html()).toMatchInlineSnapshot(`
+
+<div hidden="hidden"
+     class="ais-DynamicWidgets"
+>
+  <div class="ais-DynamicWidgets-widget">
+  </div>
+  <div class="ais-DynamicWidgets-widget">
+  </div>
+  <div class="ais-DynamicWidgets-widget">
+    <div class="ais-Panel">
+      <div class="ais-Panel-body">
+      </div>
+    </div>
+  </div>
+</div>
+
+`);
+});
+
+it('renders nothing without children', () => {
+  __setState({
+    attributesToRender: [],
+  });
+
+  const wrapper = mount(DynamicWidgets, {
+    propsData: {
+      transformItems: items => items,
+    },
+  });
+  expect(wrapper.html()).toMatchInlineSnapshot(`
+
+<div class="ais-DynamicWidgets">
+</div>
+
+`);
+});
+
+it('renders nothing with empty items', () => {
```

Comment:
Perhaps we can add a test with a panel header and empty items?

Reply:
2002f4b